receiver/redisreceiver/config.go

```diff
@@ -36,18 +36,21 @@ type Config struct {
 	TLS configtls.ClientConfig `mapstructure:"tls,omitempty"`
 
 	MetricsBuilderConfig metadata.MetricsBuilderConfig `mapstructure:",squash"`
+
+	ClusterInfoEnabled bool `mapstructure:"cluster_info_enabled"`
```

Comment:
This should be calculated by checking if any of the clusterInfo metrics are enabled.

Otherwise a user could enable the metrics, but disable this flag, and end up not getting the metrics they expect.